### PR TITLE
Architecture docs for Mephisto

### PR DESCRIPTION
**Pull request by JackUrb**
Creates some high level but moderate depth overview docs into the core details of Mephisto's architecture. Leaves some details out in the interest of space, but sets the scene such that the rest of the READMEs will relate back to this intended architecture.

If the charts aren't loading, going to either the `/img/` or `/edit/` links should give what will be rendered once this is really up.

The two that aren't showing are the simpler supervisor example [here](https://mermaid.ink/img/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gICAgcGFydGljaXBhbnQgQyBhcyBDaGFubmVsIChBcmNoaXRlY3QpXG4gICAgcGFydGljaXBhbnQgUyBhcyBTdXBlcnZpc29yXG4gICAgcGFydGljaXBhbnQgQiBhcyBCbHVlcHJpbnRcbiAgICBwYXJ0aWNpcGFudCBEQiBhcyBMb2NhbE1lcGhpc3RvREJcbiAgICBwYXJ0aWNpcGFudCBUUk4gYXMgVGFza1J1blxuICAgIHBhcnRpY2lwYW50IFRSIGFzIFRhc2tSdW5uZXIgKEJsdWVwcmludClcbiAgICBDLT4-UzogTmV3IHdvcmtlciBjb25uZWN0aW9uIFhcbiAgICBTLT4-REI6IElzIFggcmVnaXN0ZXJlZD9cbiAgICBEQi0-PlM6IE5vXG4gICAgUy0-PkRCOiBSZWdpc3RlciBYIGFzIHdvcmtlcl8xXG4gICAgUy0-PkI6IElzIHdvcmtlcl8xIHF1YWxpZmllZD9cbiAgICBCLT4-UzogWWVzIChubyBxdWFscylcbiAgICBTLT4-VFJOOiBHZXQgdmFsaWQgYXNzaWdubWVudCBmb3Igd29ya2VyXzFcbiAgICBUUk4tPj5TOiB3b3JrZXJfMSBjYW4gd29yayBvbiBhc3NpZ25tZW50XzFcbiAgICBTLT4-REI6IFJlZ2lzdGVyIEFnZW50XzEgZm9yIHdvcmtlcl8xIG9uIHVuaXRfMSBvZiBhc3NpZ25tZW50XzFcbiAgICBTLT4-VFI6IExhdW5jaCB0aGUgYXNzaWdubWVudCBmb3IgYWdlbnRfMSBvbiB1bml0XzFcbiAgICBUUi0-PlM6IEluaXRpYWxpemF0aW9uIGRhdGEgZXZlbnQgZm9yIGFnZW50XzFcbiAgICBTLT4-QzogRldEIGluaXQgZGF0YVxuICAgIExvb3AgV2hpbGUgdGFzayBub3QgZG9uZVxuICAgIEMtPj5TOiBXb3JrZXIgYWN0aW9uXG4gICAgUy0-PlRSOiBGV0Qgd29ya2VyIGFjdGlvblxuICAgIFRSLT4-UzogUmVzcG9uc2VcbiAgICBTLT4-QzogRldEIHJlc3BvbnNlXG4gICAgZW5kXG4gICAgQy0-PlM6IFRhc2sgc3VibWl0dGVkXG4gICBcbiAgICAgICAgICAgICIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0IiwidGhlbWVWYXJpYWJsZXMiOnsiYmFja2dyb3VuZCI6IndoaXRlIiwicHJpbWFyeUNvbG9yIjoiI0VDRUNGRiIsInNlY29uZGFyeUNvbG9yIjoiI2ZmZmZkZSIsInRlcnRpYXJ5Q29sb3IiOiJoc2woODAsIDEwMCUsIDk2LjI3NDUwOTgwMzklKSIsInByaW1hcnlCb3JkZXJDb2xvciI6ImhzbCgyNDAsIDYwJSwgODYuMjc0NTA5ODAzOSUpIiwic2Vjb25kYXJ5Qm9yZGVyQ29sb3IiOiJoc2woNjAsIDYwJSwgODMuNTI5NDExNzY0NyUpIiwidGVydGlhcnlCb3JkZXJDb2xvciI6ImhzbCg4MCwgNjAlLCA4Ni4yNzQ1MDk4MDM5JSkiLCJwcmltYXJ5VGV4dENvbG9yIjoiIzEzMTMwMCIsInNlY29uZGFyeVRleHRDb2xvciI6IiMwMDAwMjEiLCJ0ZXJ0aWFyeVRleHRDb2xvciI6InJnYig5LjUwMDAwMDAwMDEsIDkuNTAwMDAwMDAwMSwgOS41MDAwMDAwMDAxKSIsImxpbmVDb2xvciI6IiMzMzMzMzMiLCJ0ZXh0Q29sb3IiOiIjMzMzIiwibWFpbkJrZyI6IiNFQ0VDRkYiLCJzZWNvbmRCa2ciOiIjZmZmZmRlIiwiYm9yZGVyMSI6IiM5MzcwREIiLCJib3JkZXIyIjoiI2FhYWEzMyIsImFycm93aGVhZENvbG9yIjoiIzMzMzMzMyIsImZvbnRGYW1pbHkiOiJcInRyZWJ1Y2hldCBtc1wiLCB2ZXJkYW5hLCBhcmlhbCIsImZvbnRTaXplIjoiMTZweCIsImxhYmVsQmFja2dyb3VuZCI6IiNlOGU4ZTgiLCJub2RlQmtnIjoiI0VDRUNGRiIsIm5vZGVCb3JkZXIiOiIjOTM3MERCIiwiY2x1c3RlckJrZyI6IiNmZmZmZGUiLCJjbHVzdGVyQm9yZGVyIjoiI2FhYWEzMyIsImRlZmF1bHRMaW5rQ29sb3IiOiIjMzMzMzMzIiwidGl0bGVDb2xvciI6IiMzMzMiLCJlZGdlTGFiZWxCYWNrZ3JvdW5kIjoiI2U4ZThlOCIsImFjdG9yQm9yZGVyIjoiaHNsKDI1OS42MjYxNjgyMjQzLCA1OS43NzY1MzYzMTI4JSwgODcuOTAxOTYwNzg0MyUpIiwiYWN0b3JCa2ciOiIjRUNFQ0ZGIiwiYWN0b3JUZXh0Q29sb3IiOiJibGFjayIsImFjdG9yTGluZUNvbG9yIjoiZ3JleSIsInNpZ25hbENvbG9yIjoiIzMzMyIsInNpZ25hbFRleHRDb2xvciI6IiMzMzMiLCJsYWJlbEJveEJrZ0NvbG9yIjoiI0VDRUNGRiIsImxhYmVsQm94Qm9yZGVyQ29sb3IiOiJoc2woMjU5LjYyNjE2ODIyNDMsIDU5Ljc3NjUzNjMxMjglLCA4Ny45MDE5NjA3ODQzJSkiLCJsYWJlbFRleHRDb2xvciI6ImJsYWNrIiwibG9vcFRleHRDb2xvciI6ImJsYWNrIiwibm90ZUJvcmRlckNvbG9yIjoiI2FhYWEzMyIsIm5vdGVCa2dDb2xvciI6IiNmZmY1YWQiLCJub3RlVGV4dENvbG9yIjoiYmxhY2siLCJhY3RpdmF0aW9uQm9yZGVyQ29sb3IiOiIjNjY2IiwiYWN0aXZhdGlvbkJrZ0NvbG9yIjoiI2Y0ZjRmNCIsInNlcXVlbmNlTnVtYmVyQ29sb3IiOiJ3aGl0ZSIsInNlY3Rpb25Ca2dDb2xvciI6InJnYmEoMTAyLCAxMDIsIDI1NSwgMC40OSkiLCJhbHRTZWN0aW9uQmtnQ29sb3IiOiJ3aGl0ZSIsInNlY3Rpb25Ca2dDb2xvcjIiOiIjZmZmNDAwIiwidGFza0JvcmRlckNvbG9yIjoiIzUzNGZiYyIsInRhc2tCa2dDb2xvciI6IiM4YTkwZGQiLCJ0YXNrVGV4dExpZ2h0Q29sb3IiOiJ3aGl0ZSIsInRhc2tUZXh0Q29sb3IiOiJ3aGl0ZSIsInRhc2tUZXh0RGFya0NvbG9yIjoiYmxhY2siLCJ0YXNrVGV4dE91dHNpZGVDb2xvciI6ImJsYWNrIiwidGFza1RleHRDbGlja2FibGVDb2xvciI6IiMwMDMxNjMiLCJhY3RpdmVUYXNrQm9yZGVyQ29sb3IiOiIjNTM0ZmJjIiwiYWN0aXZlVGFza0JrZ0NvbG9yIjoiI2JmYzdmZiIsImdyaWRDb2xvciI6ImxpZ2h0Z3JleSIsImRvbmVUYXNrQmtnQ29sb3IiOiJsaWdodGdyZXkiLCJkb25lVGFza0JvcmRlckNvbG9yIjoiZ3JleSIsImNyaXRCb3JkZXJDb2xvciI6IiNmZjg4ODgiLCJjcml0QmtnQ29sb3IiOiJyZWQiLCJ0b2RheUxpbmVDb2xvciI6InJlZCIsImxhYmVsQ29sb3IiOiJibGFjayIsImVycm9yQmtnQ29sb3IiOiIjNTUyMjIyIiwiZXJyb3JUZXh0Q29sb3IiOiIjNTUyMjIyIiwiY2xhc3NUZXh0IjoiIzEzMTMwMCIsImZpbGxUeXBlMCI6IiNFQ0VDRkYiLCJmaWxsVHlwZTEiOiIjZmZmZmRlIiwiZmlsbFR5cGUyIjoiaHNsKDMwNCwgMTAwJSwgOTYuMjc0NTA5ODAzOSUpIiwiZmlsbFR5cGUzIjoiaHNsKDEyNCwgMTAwJSwgOTMuNTI5NDExNzY0NyUpIiwiZmlsbFR5cGU0IjoiaHNsKDE3NiwgMTAwJSwgOTYuMjc0NTA5ODAzOSUpIiwiZmlsbFR5cGU1IjoiaHNsKC00LCAxMDAlLCA5My41Mjk0MTE3NjQ3JSkiLCJmaWxsVHlwZTYiOiJoc2woOCwgMTAwJSwgOTYuMjc0NTA5ODAzOSUpIiwiZmlsbFR5cGU3IjoiaHNsKDE4OCwgMTAwJSwgOTMuNTI5NDExNzY0NyUpIn19LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ). 
And the complicated one [here](https://mermaid.ink/img/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gICAgcGFydGljaXBhbnQgQyBhcyBDaGFubmVsIChBcmNoaXRlY3QpXG4gICAgcGFydGljaXBhbnQgUyBhcyBTdXBlcnZpc29yXG4gICAgcGFydGljaXBhbnQgQiBhcyBCbHVlcHJpbnRcbiAgICBwYXJ0aWNpcGFudCBEQiBhcyBMb2NhbE1lcGhpc3RvREJcbiAgICBwYXJ0aWNpcGFudCBUUk4gYXMgVGFza1J1blxuICAgIHBhcnRpY2lwYW50IFRSIGFzIFRhc2tSdW5uZXIgKEJsdWVwcmludClcbiAgICBDLT4-UzogTmV3IHdvcmtlciBjb25uZWN0aW9uIFhcbiAgICBTLT4-REI6IElzIFggcmVnaXN0ZXJlZD9cbiAgICBEQi0-PlM6IFllcywgYXMgd29ya2VyXzFcbiAgICBTLT4-QjogSXMgd29ya2VyXzEgcXVhbGlmaWVkP1xuICAgIEItPj5TOiBZZXMgKHBhc3NlZCBvbmJvYXJkaW5nKVxuICAgIFMtPj5UUk46IEdldCB2YWxpZCBhc3NpZ25tZW50IGZvciB3b3JrZXJfMVxuICAgIFRSTi0-PlM6IHdvcmtlcl8xIGNhbiB3b3JrIG9uIGFzc2lnbm1lbnRfMVxuICAgIFMtPj5EQjogUmVnaXN0ZXIgYWdlbnRfMSBmb3Igd29ya2VyXzEgb24gdW5pdF8xIG9mIGFzc2lnbm1lbnRfMVxuICAgIEFjdGl2YXRlIFNcbiAgICBDLT4-UzogTmV3IHdvcmtlciBjb25uZWN0aW9uIFlcbiAgICBTLT4-REI6IElzIFkgcmVnaXN0ZXJlZD9cbiAgICBEQi0-PlM6IFllcywgYXMgd29ya2VyXzJcbiAgICBTLT4-QjogSXMgd29ya2VyXzIgcXVhbGlmaWVkP1xuICAgIEItPj5TOiBObyAobm8gb25ib2FyZGluZylcbiAgICBTLT4-VFI6IExhdW5jaCB0aGUgb25ib2FyZGluZyBmb3Igd29ya2VyXzJcbiAgICBUUi0-PlM6IE9uYm9hcmRpbmcgZGF0YSBwYWNrZXRcbiAgICBTLT4-QzogRldEIHBhY2tldCB0byB3b3JrZXJfMlxuICAgIEMtPj5TOiBDb21wbGV0ZWQgb25ib2FyZCBwYWNrZXRcbiAgICBTLT4-QjogSXMgdGhpcyBvbmJvYXJkaW5nIGNvcnJlY3RcbiAgICBCLT4-UzogTG9va3MgZ29vZFxuICAgIFMtPj5EQjogR3JhbnQgd29ya2VyXzIgcXVhbGlmaWNhdGlvblxuICAgIFMtPj5EQjogUmVnaXN0ZXIgYWdlbnRfMiBmb3Igd29ya2VyXzIgb24gdW5pdF8yIG9mIGFzc2lnbm1lbnRfMVxuICAgIERlYWN0aXZhdGUgU1xuICAgIFMtPj5UUjogTGF1bmNoIGFzc2lnbm1lbnRfMVxuICAgIFRSLT4-UzogSW5pdGlhbGl6YXRpb24gZGF0YSBldmVudHMgZm9yIGFnZW50c1xuICAgIFMtPj5DOiBGV0QgaW5pdCBkYXRhXG4gICAgQy0-PlM6IGFnZW50XzEgZGlzY29ubmVjdFxuICAgIFMtPj5DOiB3YXJuIGFnZW50XzIgb2YgZGlzY29ubmVjdFxuICAgIFMtPj5EQjogTWFyayBhZ2VudHMgYW5kIGFzc2lnbm1lbnQgaW5jb21wbGV0ZVxuICAgXG4gICAgICAgICAgIiwibWVybWFpZCI6eyJ0aGVtZSI6ImRlZmF1bHQiLCJ0aGVtZVZhcmlhYmxlcyI6eyJiYWNrZ3JvdW5kIjoid2hpdGUiLCJwcmltYXJ5Q29sb3IiOiIjRUNFQ0ZGIiwic2Vjb25kYXJ5Q29sb3IiOiIjZmZmZmRlIiwidGVydGlhcnlDb2xvciI6ImhzbCg4MCwgMTAwJSwgOTYuMjc0NTA5ODAzOSUpIiwicHJpbWFyeUJvcmRlckNvbG9yIjoiaHNsKDI0MCwgNjAlLCA4Ni4yNzQ1MDk4MDM5JSkiLCJzZWNvbmRhcnlCb3JkZXJDb2xvciI6ImhzbCg2MCwgNjAlLCA4My41Mjk0MTE3NjQ3JSkiLCJ0ZXJ0aWFyeUJvcmRlckNvbG9yIjoiaHNsKDgwLCA2MCUsIDg2LjI3NDUwOTgwMzklKSIsInByaW1hcnlUZXh0Q29sb3IiOiIjMTMxMzAwIiwic2Vjb25kYXJ5VGV4dENvbG9yIjoiIzAwMDAyMSIsInRlcnRpYXJ5VGV4dENvbG9yIjoicmdiKDkuNTAwMDAwMDAwMSwgOS41MDAwMDAwMDAxLCA5LjUwMDAwMDAwMDEpIiwibGluZUNvbG9yIjoiIzMzMzMzMyIsInRleHRDb2xvciI6IiMzMzMiLCJtYWluQmtnIjoiI0VDRUNGRiIsInNlY29uZEJrZyI6IiNmZmZmZGUiLCJib3JkZXIxIjoiIzkzNzBEQiIsImJvcmRlcjIiOiIjYWFhYTMzIiwiYXJyb3doZWFkQ29sb3IiOiIjMzMzMzMzIiwiZm9udEZhbWlseSI6IlwidHJlYnVjaGV0IG1zXCIsIHZlcmRhbmEsIGFyaWFsIiwiZm9udFNpemUiOiIxNnB4IiwibGFiZWxCYWNrZ3JvdW5kIjoiI2U4ZThlOCIsIm5vZGVCa2ciOiIjRUNFQ0ZGIiwibm9kZUJvcmRlciI6IiM5MzcwREIiLCJjbHVzdGVyQmtnIjoiI2ZmZmZkZSIsImNsdXN0ZXJCb3JkZXIiOiIjYWFhYTMzIiwiZGVmYXVsdExpbmtDb2xvciI6IiMzMzMzMzMiLCJ0aXRsZUNvbG9yIjoiIzMzMyIsImVkZ2VMYWJlbEJhY2tncm91bmQiOiIjZThlOGU4IiwiYWN0b3JCb3JkZXIiOiJoc2woMjU5LjYyNjE2ODIyNDMsIDU5Ljc3NjUzNjMxMjglLCA4Ny45MDE5NjA3ODQzJSkiLCJhY3RvckJrZyI6IiNFQ0VDRkYiLCJhY3RvclRleHRDb2xvciI6ImJsYWNrIiwiYWN0b3JMaW5lQ29sb3IiOiJncmV5Iiwic2lnbmFsQ29sb3IiOiIjMzMzIiwic2lnbmFsVGV4dENvbG9yIjoiIzMzMyIsImxhYmVsQm94QmtnQ29sb3IiOiIjRUNFQ0ZGIiwibGFiZWxCb3hCb3JkZXJDb2xvciI6ImhzbCgyNTkuNjI2MTY4MjI0MywgNTkuNzc2NTM2MzEyOCUsIDg3LjkwMTk2MDc4NDMlKSIsImxhYmVsVGV4dENvbG9yIjoiYmxhY2siLCJsb29wVGV4dENvbG9yIjoiYmxhY2siLCJub3RlQm9yZGVyQ29sb3IiOiIjYWFhYTMzIiwibm90ZUJrZ0NvbG9yIjoiI2ZmZjVhZCIsIm5vdGVUZXh0Q29sb3IiOiJibGFjayIsImFjdGl2YXRpb25Cb3JkZXJDb2xvciI6IiM2NjYiLCJhY3RpdmF0aW9uQmtnQ29sb3IiOiIjZjRmNGY0Iiwic2VxdWVuY2VOdW1iZXJDb2xvciI6IndoaXRlIiwic2VjdGlvbkJrZ0NvbG9yIjoicmdiYSgxMDIsIDEwMiwgMjU1LCAwLjQ5KSIsImFsdFNlY3Rpb25Ca2dDb2xvciI6IndoaXRlIiwic2VjdGlvbkJrZ0NvbG9yMiI6IiNmZmY0MDAiLCJ0YXNrQm9yZGVyQ29sb3IiOiIjNTM0ZmJjIiwidGFza0JrZ0NvbG9yIjoiIzhhOTBkZCIsInRhc2tUZXh0TGlnaHRDb2xvciI6IndoaXRlIiwidGFza1RleHRDb2xvciI6IndoaXRlIiwidGFza1RleHREYXJrQ29sb3IiOiJibGFjayIsInRhc2tUZXh0T3V0c2lkZUNvbG9yIjoiYmxhY2siLCJ0YXNrVGV4dENsaWNrYWJsZUNvbG9yIjoiIzAwMzE2MyIsImFjdGl2ZVRhc2tCb3JkZXJDb2xvciI6IiM1MzRmYmMiLCJhY3RpdmVUYXNrQmtnQ29sb3IiOiIjYmZjN2ZmIiwiZ3JpZENvbG9yIjoibGlnaHRncmV5IiwiZG9uZVRhc2tCa2dDb2xvciI6ImxpZ2h0Z3JleSIsImRvbmVUYXNrQm9yZGVyQ29sb3IiOiJncmV5IiwiY3JpdEJvcmRlckNvbG9yIjoiI2ZmODg4OCIsImNyaXRCa2dDb2xvciI6InJlZCIsInRvZGF5TGluZUNvbG9yIjoicmVkIiwibGFiZWxDb2xvciI6ImJsYWNrIiwiZXJyb3JCa2dDb2xvciI6IiM1NTIyMjIiLCJlcnJvclRleHRDb2xvciI6IiM1NTIyMjIiLCJjbGFzc1RleHQiOiIjMTMxMzAwIiwiZmlsbFR5cGUwIjoiI0VDRUNGRiIsImZpbGxUeXBlMSI6IiNmZmZmZGUiLCJmaWxsVHlwZTIiOiJoc2woMzA0LCAxMDAlLCA5Ni4yNzQ1MDk4MDM5JSkiLCJmaWxsVHlwZTMiOiJoc2woMTI0LCAxMDAlLCA5My41Mjk0MTE3NjQ3JSkiLCJmaWxsVHlwZTQiOiJoc2woMTc2LCAxMDAlLCA5Ni4yNzQ1MDk4MDM5JSkiLCJmaWxsVHlwZTUiOiJoc2woLTQsIDEwMCUsIDkzLjUyOTQxMTc2NDclKSIsImZpbGxUeXBlNiI6ImhzbCg4LCAxMDAlLCA5Ni4yNzQ1MDk4MDM5JSkiLCJmaWxsVHlwZTciOiJoc2woMTg4LCAxMDAlLCA5My41Mjk0MTE3NjQ3JSkifX0sInVwZGF0ZUVkaXRvciI6ZmFsc2V9).

If these look good I'll save the images and put them in manually as I think it's a github limit on transferring data via url.

### CODE REVIEW

File: docs/architecture_overview.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+*Annotate [these] images* would be an example of a *Task*, as would *Translate  [these] sentences into [language]*. Each of these could be extended, for example by giving multiple (similar) annotation types or translating into many different languages. These two tasks though could not reasonably be combined into one *Task* though, as they are using different context. 
+
+Methods of the `Task` class surround being able to consolidate information about a specific *Task*, and the task class is uniquely indexed by the `task_name` property. 
+
+### Task Runs
+A task run can be thought of as part of running an overall task. So, if you wanted to annotate a set of images, but could only do it in parts, each of those parts would be a `TaskRun` of the image annotation `Task` that you are working on. Each `TaskRun` is provided its own set of configuration, and as such your `Task` to translate some number of sentences into different languages may have a `TaskRun` that asks workers to translate to Spanish while another `TaskRun` asks workers to translate the same sentences to French.
```
Comment (saraEbrahim) [inline docs/architecture_overview.md:29]: It'd be wonderful if the previous example was extended with numbers (e.g. if you need to classify 100 images 3 times) how many `TaskRun` Mephisto will have, or just explain what does (parts) mean in this sentence (could classify images but in parts)

File: docs/architecture_overview.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+### Assignment
+An `Assignment` can be thought of the minimal subgoal of a given `Task`. If your `Task` is to *Annotate [these] images*, one such subgoal could be to *Annotate [this] image*. It may be appropriate to batch these, such that the assignment is actually to *Annotate [these 5] images*, but the goal here is to ensure that an `Assignment` represents a reasonable amount of work to accomplish, but small enough to be reviewed as a whole.
+
+In some cases, an `Assignment` may relate to something that needs to be done by multiple workers at once, such as a cooperative task. In other cases, an `Assignment` may relate to wanting multiple different workers to attempt the same task in order to get consensus. We refer to this as a task being *concurrent* or not. In either case, however, Mephisto will prevent any given `Worker` from participating in any `Assignment` more than once.
```
Comment (saraEbrahim) [inline docs/architecture_overview.md:46]: this is where an example with numbers would really help to understand what `Assignment`

File: docs/architecture_overview.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+### Unit
+The `Unit` is the smallest bit of work that is part of a task, in that it comprises a single worker's contribution to that task. It has the required information for initializing a specific instance of work to the worker, and is responsible for having up-to-date information about the real-life status of that work.
+
+Some `Assignment`'s may have just one `Unit`, should the task be something that can be completable by just one worker and only needs completing once. Some `Assignment`'s, like a conversation, may be comprised of two (or more) `Unit`'s, each representing one worker's place in the conversation.
```
Comment (saraEbrahim) [inline docs/architecture_overview.md:50]: this is for image labeling as an example right?

**Review by saraEbrahim — APPROVED** [review]
This is a great effort. Thanks for clarifying Mephisto in a nice way!

File: docs/architecture_overview.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+Everything in the data model is backed inside of the `MephistoDB`, such that other processes can examine the status of tasks, view worker behavior, or access the information long after there is no active process running a crowdsourcing task. 
+## Runs
```
[the remark this answers is not in the document]
Comment (pringshia) [reply docs/architecture_overview.md:18]: +1! love this

File: docs/architecture_overview.md (hunk 3 above)
Comment (pringshia) [inline docs/architecture_overview.md:50]: How about in a consensus situation? Is it similar in that a single `Assignment` will have multiple `Unit`s assigned concurrently?

File: docs/architecture_overview.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+### Worker
+The Mephisto `Worker` object keeps track of a specific person after they've interacted with Mephisto. The `Worker` class is able to track overall pay, acceptance rates, and other trends based on the data that is linked to it. A `Worker` can also be granted qualifications. Mephisto uses the `Worker` class primarily to determine eligibility for specific `Unit`'s, and it exposes other functions as helpers for analysis.
+### Agent
+The `Agent` object can be thought of as an instance of a `Worker` paired to a specific `Unit`, in that it tracks the work of the human on the other end as they complete the `Unit`. It is responsible for tracking the state of the work done, as well as saving the data associated with the work. Over time, a `Worker` that continues to do tasks will end up having many `Agent`'s.
```
Comment (pringshia) [inline docs/architecture_overview.md:57]: Is it safe to say that a Worker will have as many Agents as Units they have worked on?

File: docs/architecture_overview.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+
+
+Each of these classes can be considered themselves as generalizations which can be configured for usage in specific instances. As such, each of these classes provides a list of parameters that are used to configure their behaviors, and subclasses can provide additional parameters for additional specification. No `Blueprint` should only work for one specific tasks, but it should be a blueprint from which many similar tasks can be built.
+
+The classes are all expected to provide an `Args` class, which is a dataclass that we can use as a structured config alongside [Hydra](https://hydra.cc/) to populate configuration. It should also define an `assert_task_args` method, which ensures that given a set of configuration, a run with that particular configuration will not fail to execute for any configuration related reasons.
+
+There's also the abstraction of the `MephistoDB`, which defines the data operations that Mephisto requires to operate properly. If a specific setup requires specialized data handling, any class implementing this interface could stand in for it.
```
Comment (pringshia) [inline docs/architecture_overview.md:74]: Should we also list MephistoDB as a core abstraction (making the 3 abstractions into 4?). It's mentioned in the abstractions section here so just wondering.

File: docs/architecture_overview.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+### BlueprintArgs
+These define the specific arguments for configuring a blueprint. Some base arguments relevant to all blueprints are provided already, and classes that override `BlueprintArgs` can add anything else they want.
+### SharedTaskState
+Some tasks require additional state that cannot be considered static configuration, such as validation functions for accurate completion of a task, or a counter for the number of tasks completed in a run so far, or any other python state that multiple different assignments may be interested in accessing or altering. This class is for the edge cases that need something more than static configuration.
```
Comment (pringshia) [inline docs/architecture_overview.md:81]: Perhaps it's worth mentioning here if this is sort of a singleton state that all tasks have access to, or does each assignment/unit get it's own instance?

File: docs/architecture_overview.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+### TaskRunner
+Likewise, all tasks need to have some kind of backend logic that dictates the kind of information we intend to receive. For simple 'static' tasks where we give the worker some info and they give us their work, this may be just a single exchange, however more complicated tasks can put any other kind of difficult flow control here.
+### AgentState
+As different tasks will save different data, Blueprints need a way to specify what kind of data is being saved. The `AgentState` is responsible for parsing out the required information that will need to be saved during a task, viewed during review, and exported during a final compilation. It receives both the initial information given to the worker, as well as every action packet sent to or by the worker.
+
+-------------
+### Onboarding
+Blueprints also provide simple hooks for "onboarding" tasks, which can be considered as tasks that a worker must complete in order to qualify for the real thing. These hooks are added to the `TaskRunner` and `AgentState` to provide a way to set the flow for onboarding, what is saved, and how the work is evaluated.
```
Comment (pringshia) [inline docs/architecture_overview.md:91]: > which can be considered as tasks that a worker must complete

Can we replace the overloaded "tasks" here with something like `Assignment`s?

File: docs/architecture_overview.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+-------------
+### Onboarding
+Blueprints also provide simple hooks for "onboarding" tasks, which can be considered as tasks that a worker must complete in order to qualify for the real thing. These hooks are added to the `TaskRunner` and `AgentState` to provide a way to set the flow for onboarding, what is saved, and how the work is evaluated. 
+
+## Architects
+`Architect`'s contain the logic surrounding deploying a server that workers will be able to access. In many cases Mephisto is being run on compute clusters that aren't directly addressable, or in different configurations between collaborators. Mephisto should be able to run a task from start to finish regardless of the server configuration a user would like to use, and `Architect`'s provide this capability.
+
+The `Architect` class is responsible for providing Mephisto with lifecycle functions for preparing, deploying, and shutting down a given server. It's also responsible with providing access to the user via a `Channel`, which defines an interface of callbacks for incoming messages and a function for outgoing messages.
```
Comment (pringshia) [inline docs/architecture_overview.md:96]: Might be worth adding another paragraph to the end of this as such:

"You can pick from a suite of pre-configured Architects, or create/customize your own."

File: docs/architecture_overview.md (hunk 6 above)
Comment (JackUrb) [reply docs/architecture_overview.md:74]: It's hard to know - I wouldn't say that the `MephistoDB` is a clean enough abstraction for me to list it as such, and further it's not one of the ones that Mephisto knows how to auto-populate. Much of mephisto code currently depends on the LocalMephistoDB specifically.

Once I get a chance to clean up that abstraction, I'll likely list it directly with the rest.

**Pull request merged**